### PR TITLE
prevent initial focus trap while has a focused element.

### PR DESCRIPTION
**Pull request by suneox**
### Fixed Issues

$ https://github.com/Expensify/App/issues/43662
$ https://github.com/Expensify/App/issues/44956
$ https://github.com/Expensify/App/issues/44745
PROPOSAL: https://github.com/Expensify/App/issues/43662#issuecomment-2198089476

### Tests
Same QA Steps

- [x] Verify that no errors appear in the JS console

### QA Steps

* **Issue 43662:**

1. Go to staging.new.expensify.com
2. Go to chat
3. Go to + > Submit expense > Manual
4. Enter amount
5. Hit Enter
6. Verify the back icon on RHP will not have blue border selection

* **Issue 44745:**

1. Open app
2. From global menu click on 'Fab' > start chat
3. From the bottom of the page click on the referral banner
4. Verify Animation should correct when RHP page for referral is opened

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

**Issue 43662:**

https://github.com/Expensify/App/assets/11959869/57be58f4-c099-426c-a73c-d241b7f5269a

**Issue 44745:**

https://github.com/Expensify/App/assets/11959869/095399f0-4adc-4d6f-b46c-8a92ce4641bc

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/11959869/ccb59745-d1eb-4a92-98e9-471096ed1ac3

</details>

<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/11959869/10792ddb-c49d-4a5a-870f-5307af0da26e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/11959869/49089e83-f742-41fa-be7e-d717c2ef3ec8

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/11959869/5eae2c8f-e453-44d2-81db-755be60507c0

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/11959869/b9925d7b-1423-44a9-8bcd-894402d7c57b

</details>

### CODE REVIEW

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 1]
```diff
@@ -51,12 +52,13 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
                 trapStack: sharedTrapStack,
                 allowOutsideClick: true,
                 fallbackFocus: document.body,
-                // We don't want to ovverride autofocus on these screens.
-                initialFocus: () => {
-                    if (screensWithAutofocus.includes(activeRouteName)) {
+                delayInitialFocus: CONST.ANIMATED_TRANSITION,
+                // We don’t want to override autofocus while there is a focused element.
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:56]:
```suggestion
```
Comment not needed

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 2]
```diff
@@ -51,12 +52,13 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
                 trapStack: sharedTrapStack,
                 allowOutsideClick: true,
                 fallbackFocus: document.body,
-                // We don't want to ovverride autofocus on these screens.
-                initialFocus: () => {
-                    if (screensWithAutofocus.includes(activeRouteName)) {
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:56]: Remove `screensWithAutofocus`

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 3]
```diff
@@ -51,12 +52,13 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
...
+                delayInitialFocus: CONST.ANIMATED_TRANSITION,
+                // We don’t want to override autofocus while there is a focused element.
+                initialFocus: (containers) => {
+                    const hasFocusedElement = containers?.some((container) => container.contains(document.activeElement));
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:58]:
```suggestion
                    const isFocusedElementWithinFocusTrapContainer = containers?.some((container) => container.contains(document.activeElement));
```

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 4]
```diff
@@ -51,12 +52,13 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
...
+                    if (hasFocusedElement) {
                         return false;
                     }
-                    return undefined;
                 },
                 setReturnFocus: (element) => {
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:63]: Remove `setReturnFocus`

File: patches/focus-trap+7.5.4.patch [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+--- a/node_modules/focus-trap/dist/focus-trap.esm.js
++++ b/node_modules/focus-trap/dist/focus-trap.esm.js
+@@ -100,8 +100,8 @@ var isKeyForward = function isKeyForward(e) {
+ var isKeyBackward = function isKeyBackward(e) {
+   return isTabEvent(e) && e.shiftKey;
+ };
+-var delay = function delay(fn) {
+-  return setTimeout(fn, 0);
++var delay = function delay(fn, delayTime = 0) {
++  return setTimeout(fn, delayTime);
```
Comment (s77rt) [inline patches/focus-trap+7.5.4.patch:12]: I think we need to keep the extra setTimeout here and add our timeout. After testing I found out that document.activeElement does not return the correct focused element.

```suggestion
+  return setTimeout(() => setTimeout(fn, delayTime), 0);
```

File: patches/focus-trap+7.5.4.patch [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+diff --git a/node_modules/focus-trap/index.d.ts b/node_modules/focus-trap/index.d.ts
+index 400db1b..e6d1bc3 100644
+--- a/node_modules/focus-trap/index.d.ts
++++ b/node_modules/focus-trap/index.d.ts
+@@ -118,7 +118,7 @@ declare module 'focus-trap' {
+      * Setting this option to `undefined` (or a function that returns `undefined`)
+      * will result in the default behavior.
+      */
+-    initialFocus?: FocusTargetOrFalse | undefined | (() => void);
++    initialFocus?: FocusTargetValueOrFalse | ((containers?: HTMLElement[]) => FocusTargetValueOrFalse | undefined);
```
Comment (s77rt) [inline patches/focus-trap+7.5.4.patch:43]: Why we have this change `FocusTargetOrFalse` -> `FocusTargetValueOrFalse`

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx (hunk 4 above)
Comment (s77rt) [reply src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:63]: It's used to return the focus to the element that opened the modal, e.g. if you press button1 that opens a RHP, after closing the RHP, button1 should be focused again. We should follow the same principle here, that is to refocus button1 as long as we don't have another focused element.

File: patches/focus-trap+7.5.4.patch (hunk 5 above)
Comment (s77rt) [reply patches/focus-trap+7.5.4.patch:12]: Can you please elaborate what does this has to do with focus deactivation? What I am referring to is the fact that if you log `document.activeElement` you will not get the correct focused element as we are doing this too early


https://github.com/Expensify/App/assets/16493223/bc39cb75-1f9e-4ed1-9d78-302d3ffb05a1

File: patches/focus-trap+7.5.4.patch (hunk 6 above)
Comment (s77rt) [reply patches/focus-trap+7.5.4.patch:43]: Let's update `FocusTargetOrFalse`

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 7]
```diff
@@ -37,11 +32,11 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
...
-    useFocusEffect(
-        useCallback(() => {
-            activeRouteName = route.name;
-        }, [route]),
-    );
+    const mountedRef = useRef(false);
+
+    useEffect(() => {
+        mountedRef.current = true;
+    }, []);
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:39]: Revert

File: src/components/MoneyRequestConfirmationList.tsx [hunk 8]
```diff
@@ -726,6 +726,16 @@ function MoneyRequestConfirmationList({
...
 
+    const confirmButtonRef = useRef<View>(null);
+
+    useFocusEffect(
+        useCallback(() => {
+            InteractionManager.runAfterInteractions(() => {
+                confirmButtonRef.current?.focus();
+            });
+        }, []),
+    );
```
Comment (s77rt) [inline src/components/MoneyRequestConfirmationList.tsx:737]: use `useEffect` only and avoid `InteractionManager.runAfterInteractions`

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by mountiny** [comment]
@suneox @s77rt can you please confirm this PR should fix this issue as well? https://github.com/Expensify/App/issues/44956 if yes, please add it to the PR description and add the necessary test steps to the coverage, thank you!

**Comment by s77rt** [comment]
@mountiny Yes it will fix that issue

File: src/components/MoneyRequestConfirmationList.tsx [hunk 9]
```diff
@@ -726,6 +726,12 @@ function MoneyRequestConfirmationList({
         ],
     );
 
+    const confirmButtonRef = useRef<View>(null);
+
+    useEffect(() => {
+        confirmButtonRef.current?.focus();
+    }, []);
+
```
Comment (s77rt) [inline src/components/MoneyRequestConfirmationList.tsx:734]: The focus should be delayed not to disrupt the animation. Please use `useAutoFocusInput`

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 10]
```diff
@@ -51,18 +40,12 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
...
                 allowOutsideClick: true,
                 fallbackFocus: document.body,
-                // We don't want to ovverride autofocus on these screens.
-                initialFocus: () => {
-                    if (screensWithAutofocus.includes(activeRouteName)) {
-                        return false;
-                    }
-                    return undefined;
-                },
-                setReturnFocus: (element) => {
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:61]: We should not return focus to the initial screen if we already have a focused element, e.g. in ReportScreen after closing RHP the Composer should remain focused


https://github.com/Expensify/App/assets/16493223/3f3cb5cd-8c77-49cc-a948-b6675666d024

File: src/components/FocusTrap/FocusTrapForScreen/FocusTrapProps.ts [hunk 11]
```diff
@@ -1,5 +1,6 @@
 type FocusTrapForScreenProps = {
     children: React.ReactNode;
+    testID?: string;
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/FocusTrapProps.ts:3]: Is this change required here?

File: patches/focus-trap+7.5.4.patch (hunk 5 above)
Comment (s77rt) [reply patches/focus-trap+7.5.4.patch:12]: The correct timing is 300ms (`CONST.ANIMATED_TRANSITION`). `CONST.ANIMATION_IN_TIMING` is not related to the screen transition.

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx (hunk 10 above)
Comment (s77rt) [reply src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:62]: Yes!

Comment (s77rt) [reply src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:62]: Works for me but let's also check if `document.activeElement` is `null`. In that case we want to return the focus as well

File: patches/focus-trap+7.5.4.patch (hunk 5 above)
Comment (s77rt) [reply patches/focus-trap+7.5.4.patch:12]: Given that this already had it's own `setTimeout` even when no timeout is actually required, we should just add another `setTimeout`. The former is to fix whatever the library was trying to fix and the later is for our actual wanted delay.

File: patches/focus-trap+7.5.4.patch [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+ 
+   type MouseEventToBoolean = (event: MouseEvent | TouchEvent) => boolean;
+   type KeyboardEventToBoolean = (event: KeyboardEvent) => boolean;
+@@ -147,7 +147,8 @@ declare module 'focus-trap' {
+     setReturnFocus?:
+       | FocusTargetValueOrFalse
+       | ((
+-          nodeFocusedBeforeActivation: HTMLElement | SVGElement
++          nodeFocusedBeforeActivation: HTMLElement | SVGElement,
++          containers: Array<HTMLElement>
```
Comment (s77rt) [inline patches/focus-trap+7.5.4.patch:71]: Same ^

File: patches/focus-trap+7.5.4.patch [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
++    delayInitialFocus?: boolean | number;
+     /**
+      * Default: `window.document`. Document where the focus trap will be active.
+      * This allows to use FocusTrap in an iFrame context.
+@@ -222,6 +223,8 @@ declare module 'focus-trap' {
+      * keyboard navigation within the trap, for example. Also see `isKeyForward()` option.
+      */
+     isKeyBackward?: KeyboardEventToBoolean;
++
++    testID?: string;
```
Comment (s77rt) [inline patches/focus-trap+7.5.4.patch:89]: Not needed

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 14]
```diff
@@ -52,15 +40,16 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
                 trapStack: sharedTrapStack,
                 allowOutsideClick: true,
                 fallbackFocus: document.body,
-                // We don't want to ovverride autofocus on these screens.
-                initialFocus: () => {
-                    if (screensWithAutofocus.includes(activeRouteName)) {
+                delayInitialFocus: CONST.ANIMATED_TRANSITION + CONST.ANIMATION_IN_TIMING,
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:43]:
```suggestion
                delayInitialFocus: CONST.ANIMATED_TRANSITION,
```

File: src/components/MoneyRequestConfirmationList.tsx [hunk 15]
```diff
@@ -757,6 +760,9 @@ function MoneyRequestConfirmationList({
             />
         ) : (
             <ButtonWithDropdownMenu
+                buttonRef={(ref) => {
+                    inputCallbackRef(ref);
+                }}
```
Comment (s77rt) [inline src/components/MoneyRequestConfirmationList.tsx:765]:
```suggestion
                buttonRef={inputCallbackRef}
```

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/FocusTrap/FocusTrapForScreen/index.web.tsx [hunk 16]
```diff
@@ -52,15 +40,16 @@ function FocusTrapForScreen({children}: FocusTrapProps) {
                 trapStack: sharedTrapStack,
                 allowOutsideClick: true,
                 fallbackFocus: document.body,
-                // We don't want to ovverride autofocus on these screens.
-                initialFocus: () => {
-                    if (screensWithAutofocus.includes(activeRouteName)) {
+                delayInitialFocus: CONST.ANIMATED_TRANSITION,
+                initialFocus: (focusTrapContainers) => {
+                    const hasFocusedElementInsideContainer = focusTrapContainers?.some((container) => container.contains(document.activeElement));
```
Comment (s77rt) [inline src/components/FocusTrap/FocusTrapForScreen/index.web.tsx:45]:
```suggestion
                    const isFocusedElementInsideContainer = focusTrapContainers?.some((container) => container.contains(document.activeElement));
```

NAB

**Comment by s77rt** [comment]
Bug: After pressing Enter selecting the participant, the back button is focused initially 



https://github.com/user-attachments/assets/8a34e98b-9a3a-49fb-bb20-b8e6fdbaa298

**Comment by s77rt** [comment]
@suneox We can't use `ANIMATION_IN_TIMING` as it's unrelated to screen transition it's for the popover opening duration. We need to figure out another solution for this

**Comment by s77rt** [comment]
@suneox How this works with the search input? Let's follow the same approach there


https://github.com/user-attachments/assets/8c93535a-2678-4bcc-a096-28ca771f4d58

**Comment by s77rt** [comment]
@suneox What I meant is to use the same focus approach used in the participants selection and use it in the confirmation page (do not use useAutoFocusInput)

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


Build issue
</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/user-attachments/assets/d0b5e8c8-fd96-4bd7-9555-c33a5a3bf652

</details>

<details>
<summary>iOS: Native</summary>



https://github.com/user-attachments/assets/615f286d-e24c-4223-a230-0ffc574fcfb3


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/df1778be-110f-4d65-9e42-fabdc45c1e6e


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/9a809d6e-b813-4143-ad55-81ed4a6d907e


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/user-attachments/assets/9daf6fbf-cfa1-422b-9a23-02a830a1b0d3


</details>

**Review by s77rt — APPROVED** [review]

**Review by chiragsalian — APPROVED** [review]
LGTM

**Comment by akinwale** [comment]
This PR caused a [regression](https://github.com/Expensify/App/issues/45570).

The button has the `pressOnEnter` prop set to true, so manual focus is not required.

**Pull request merged**